### DISCUSSION TITLE
Tryng to create a new repo but get an error message !

### DISCUSSION THREAD

**Discussion by Jacques2101** [community/community#56682] [New to GitHub]
Hi i am new in GitHub and I would like to create a repo for streamlit using. 
I tried to input my code and my data inside a repo but get an error message: 
```Yowza, that’s a big file. Try again with a file smaller than 25MB.```

I suppose my data file are too large to enter GitHub (they are parquet files with respectively 38.6Mo, 37.6Mo, 21.2Mo, 390Ko, 90ko and 25Ko...but I read that GitHub can support 1Go of file. 
What should I do ? Maybe I did something wrong. 

Any help will be helpful. 
thx 
<img width="1249" alt="Capture d’écran 2023-05-29 à 13 26 57" src="https://github.com/community/community/assets/49912455/07f200e3-190d-4d84-a037-806e0fa4de54">

**Comment by airtower-luna** [comment]
I'm afraid wherever you read about the 1GB limit was wrong. :sweat_smile: 

See [About large files on GitHub](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github) for details, the 25 MB limit is mentioned there (as well as others).

**Comment by anderscain63** [comment]
@Jacques2101 **To address this issue, consider the following options:**

- Git Large File Storage (Git LFS): If you have large files that need to be included in your repository, you can use Git LFS. Git LFS allows you to store large files externally while keeping lightweight pointers to those files in your repository. This helps manage version control efficiently. You can find more information about Git LFS on the official GitHub documentation: https://git-lfs.github.com/

- Ignore large files: If the large files are not essential to your repository, you can add them to the .gitignore file. This will prevent them from being tracked and uploaded to GitHub. You can specify file extensions or file patterns to be ignored.

- Compress or reduce file size: If your data files are in a format that can be compressed, consider compressing them using tools like ZIP or GZIP. This can significantly reduce the file size and make it easier to work with on GitHub.

- Host data files elsewhere: If your data files are too large to be hosted directly on GitHub, you can consider hosting them on a separate file hosting service or cloud storage platform. You can then provide instructions or a link in your repository's README file on how to obtain the data files from the external source.

Remember to assess the purpose and necessity of including large files in your repository. Large binary files or data files might be more suited for dedicated data storage platforms rather than version control repositories.

If you're still having trouble or need more guidance, please provide more specific details about the files and your intended use, and I'll be happy to assist you further.

**Comment by Jacques2101** [comment]
"Host data files elsewhere: If your data files are too large to be hosted directly on GitHub, you can consider hosting them on a separate file hosting service or cloud storage platform. You can then provide instructions or a link in your repository's README file on how to obtain the data files from the external source."
I tried to store my data on google and share it but then how to read it on Python ? 
I get an https address and don't know how to use it 

Any help will be great. 
thx.

**Reply by airtower-luna** [reply]
That really depends on how you want to use the data. One obvious option would be to tell the user to download the file(s) and place them in a certain location. Another would be to automate the download (you should probably ask for user confirmation first, or only offer it as an option), to make HTTPS requests from Python check out the [`requests` module](https://requests.readthedocs.io/en/latest/). Considering how big your files are you should not download them on every run, just once to store them locally. :sweat_smile:
